Assets/Scripts/DayNightCycle/DayAndNightControl.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Animations;
+using UnityEngine.Rendering.Universal;
+using UnityEditor;
+using Assets.Scripts.Services;
+using Assets.Scripts.Settings;
+
+namespace Assets.Scripts.DayNightCycle
+{
+    /// <summary>
+    /// The DayAndNightControl class controls the time of day and lighting of AoTTG2. Gradients for ambient light can be found in TimeCycleProfile.cs
+    /// </summary>
+    public class DayAndNightControl : MonoBehaviour
+    {
+        public GameObject Moon;
+        public ReflectionProbe ReflectionProbe;
+        public float SunTilt = -15f;
+        [SerializeField] private TimecycleProfile timecycle = null;
+        [SerializeField] private float sunRotationOffset = 0f;
+        [Tooltip("The amount of frames to wait before doing the next lighting update")]
+        [SerializeField] private int lightingUpdateInterval = 10;
+        public Material SkyboxMaterial;
+
+        [Range(0f, 24f)] public float CurrentTime;
+        public float CurrentTimeScale => CurrentTime / 24;
+        public Camera MoonCamera = null;
+        public Camera MainCamera = null;
+        public int CurrentDay = 0;
+        public Light DirectionalLight;
+        public float DayLength; 
+        public bool Pause { get; set; }
+        public float LightIntensity; //static variable to see what the main light's insensity is in the inspector
+
+        private int frames;
+       
+        void Start()
+        {
+            Pause=true;
+            Service.Settings.OnTimeSettingsChanged += Settings_OnTimeSettingsChanged;
+            //MC loads settings on start because these values need to have a default.
+            if (PhotonNetwork.isMasterClient)
+            {
+                GameSettings.Time.currentTime = CurrentTime;
```

Comment:
These are either supplied by the Difficult or the users custom settings. Because of the current implementation: the day night cycle cannot be paused, start with a specific time or specify what the length of a day should be by a global, gamemode or map setting. Remove such limitation to the fancy system.

Reply:
Sounds good, at the moment it is sort of relegated to map setting, except via the prefab, so I can see if there is a place to set map settings and make it so the prefab allows the editing of those settings to for ease of use for mappers.

Reply:
Should still be via the TimeSetting, as otherwise it won't work well for custom maps, nor for my "Lithruns Hardcore Capture Gamemode Night 24/7 Edition" difficulty setting.

Reply:
I see yeah, we can discuss the implementation over VC sometime.